Kafka/src/main/java/io/deephaven/kafka/publish/PublishToKafka.java

```diff
@@ -0,0 +1,138 @@
+package io.deephaven.kafka.publish;
+
+import io.deephaven.base.verify.Assert;
+import io.deephaven.db.tables.Table;
+import io.deephaven.db.v2.DynamicTable;
+import io.deephaven.db.v2.InstrumentedShiftAwareListenerAdapter;
+import io.deephaven.db.v2.StreamTableTools;
+import io.deephaven.db.v2.sources.chunk.Attributes;
+import io.deephaven.db.v2.sources.chunk.ObjectChunk;
+import io.deephaven.db.v2.sources.chunk.WritableObjectChunk;
+import io.deephaven.db.v2.utils.Index;
+import io.deephaven.db.v2.utils.OrderedKeys;
+import io.deephaven.db.v2.utils.ReadOnlyIndex;
+import io.deephaven.util.annotations.ReferentialIntegrity;
+import org.apache.kafka.clients.producer.RecordMetadata;
+import org.jetbrains.annotations.NotNull;
+
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.clients.producer.KafkaProducer;
+
+import java.util.Properties;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+
+public class PublishToKafka<K, V> {
+    static final int CHUNK_SIZE = 2048;
+
+    @NotNull
+    private final Table table;
+    private final KafkaProducer<K, V> producer;
+    private final String topic;
+    private final KeyOrValueSerializer<K> keySerializer;
+    private final KeyOrValueSerializer<V> valueSerializer;
+
+    @ReferentialIntegrity
+    private final PublishListener publishListener;
+
+    public PublishToKafka(final Properties props,
```

Comment:
I think we want a partition number too;
The issue of publishing subsets of keys by partition can depend on the use for defining tables with proper filtering per key; but the mapping to particular partition seems missing.

Reply:
You don't think that should be up to the partitioner specified in properties?
https://jaceklaskowski.gitbooks.io/apache-kafka/content/kafka-producer-internals-DefaultPartitioner.html

Reply:
You are right, I completely forgot about how partitions work on the publishing side.  Nevermind.